### PR TITLE
Kerberos authentication

### PR DESCRIPTION
**Pull request by tttech-schueller**
Hi!
This implements kerberos authentication as we have this setup in our company and I try to avoid typing passwords on the commandline :-)

I tried to implement that (password AND username) OR kerberos is provided as command line arguments but I failed to do this with argparse groups so I did this with an execption I hope that's also fine for you.

Additionally I forced SSL certificate check _by default_ ...

### CODE REVIEW

**Comment by allburov** [comment]
Hey, thank you for your contribution! 

Shouldn't we add requests_kerberos in the requirements? Otherwise catching `ImportError` exceptions and install all packages by hand will be kinda annoying :) 
https://github.com/devopshq/artifactory-du/blob/develop/artifactory_du/requirements.txt

File: artifactory_du/artifactory_du.py [hunk 1]
```diff
@@ -22,19 +22,28 @@ def init_logging():
...
+    :param verify: Boolean if SSL certificate should be checked
     :param aql_query_dict:
     :param max_depth_print:
     :param human_readable:
     :param all:
     :return:
     """
-    aql = ArtifactoryPath(artifactory_url, auth=(username, password), verify=False)
+    if kerberos:
+        from requests_kerberos import HTTPKerberosAuth, DISABLED
```
Comment (allburov) [inline artifactory_du/artifactory_du.py:40]: Let's move import to the top after adding the library to requirements.txt

Comment (tttech-schueller) [reply artifactory_du/artifactory_du.py]: I wasn't sure if this should be a default dependency for all installations.

**Comment by allburov** [comment]
CI failed on python3.6 :( 
Could you try to add `setuptools-rust` to requirements.txt too pls?

**Comment by tttech-schueller** [comment]
> CI failed on python3.6 :(
> Could you try to add `setuptools-rust` to requirements.txt too pls?

sure, I just saw the build error later that you did :-D

**Comment by allburov** [comment]
>I should not create the missing .pre-commit-config.yaml right?

No, I'll do it later. May be :)

Thank you!

**Pull request merged**